### PR TITLE
job-list: do not assume alloc event context always contains annotations

### PR DESCRIPTION
**Pull request by chu11**
Per discussion in #4906

I only added a regression test, no additional wider coverage tests yet per discussion in #4906.

Hopefully will have those wider extra coverage tests in soon.  But just in case that extra wider coverage can't be completed in a timely manner, just wanted to make sure we atleast get this fix in before the next release, we can always add the extra tests later.

### CODE REVIEW

**Comment by grondo** [comment]
Sounds good to me!

**Comment by chu11** [comment]
added a coverage test on top, instead of the dump file or jobtap plugin approach as mentioned in #4904, I just "hand" wrote an eventlog over an existing one.  That way we can easily add extra events in the future (vs the dump file approach).

File: t/issues/t4906-job-list-alloc-bypass.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -e
```
Comment (grondo) [inline t/issues/t4906-job-list-alloc-bypass.sh:1]: Since you have the follow on commit which adds a regression test to the job-list module, maybe this separate regression test isn't required? The reason I ask is that the t4000-issues test runs all the reproducers serially and at this point is really slow, so avoiding another `flux start` test therein would keep the testsuite a little leaner.

Comment (chu11) [reply t/issues/t4906-job-list-alloc-bypass.sh:1]: actually the follow on test doesn't test it perfectly, as the regression that tests the event `alloc {"bypass":1}`, whereas the new test added `alloc {"annotations}:<something>, "etc": 1}`.

But I could tweak it accordingly.

Comment (grondo) [reply t/issues/t4906-job-list-alloc-bypass.sh:1]: Ah, that might be good just to save some unnecessary work for the issues test driver.

Comment (chu11) [reply t/issues/t4906-job-list-alloc-bypass.sh:1]: ahhh this was a good idea, ended up there was a corner case with exception notes (which are optional).

**Comment by chu11** [comment]
re-pushed, removing the regression test and adding a new test in t2260-job-list.t instead.  Decided to add another test to check for optional context fields.  The only other one I found was userid/note in exceptions, and it ended up there was a corner case in job-list for that case, so fixed that too.

**Review by grondo — APPROVED** [review]
This LGTM. Nice job catching the exception note corner case!
